tests/test_dcrAssembleCoadd.py

```diff
@@ -0,0 +1,295 @@
+import scipy.ndimage.interpolation
+from scipy.ndimage.morphology import binary_dilation as dilate
+import unittest
+
+from lsst.afw.coord import Coord, IcrsCoord, Observatory, Weather
+import lsst.afw.geom as afwGeom
+from lsst.afw.geom import Angle, makeCdMatrix, makeSkyWcs
+import lsst.afw.image as afwImage
+import lsst.utils.tests
+from lsst.pipe.tasks.dcrAssembleCoadd import DcrAssembleCoaddTask, DcrAssembleCoaddConfig
+
+
+class DcrAssembleCoaddTestTask(lsst.utils.tests.TestCase):
+    """! A test case for the DCR-aware image coaddition algorithm.
+    """
+
+    def setUp(self):
+        self.config = DcrAssembleCoaddConfig()
+        self.config.dcrNSubbands = 3
+        self.config.lambdaEff = 478.
+        self.config.filterWidth = 147.
+        self.config.filterName = 'g'
+        badMaskPlanes = self.config.badMaskPlanes[:]
+        badMaskPlanes.append("CLIPPED")
+        size = 40
+        psfSize = 2
+        nSrc = 5
+        seed = 5
+        self.randGen = np.random
+        self.randGen.seed(seed)
+        noiseLevel = 5
+        sourceSigma = 20.
+        fluxRange = 2.
+        edgeDist = self.config.bufferSize
+        xLoc = self.randGen.random(nSrc)*(size - 2*edgeDist) + edgeDist
+        yLoc = self.randGen.random(nSrc)*(size - 2*edgeDist) + edgeDist
+        self.dcrModels = []
+        self.bbox = afwGeom.Box2I(afwGeom.Point2I(0, 0), afwGeom.Extent2I(size, size))
+        self.pixelScale = 0.2*afwGeom.arcseconds
+
+        imageSum = np.zeros((size, size))
+        for subfilter in range(self.config.dcrNSubbands):
+            flux = (self.randGen.random(nSrc)*(fluxRange - 1.) + 1.)*sourceSigma*noiseLevel
+            model = afwImage.MaskedImageF(self.bbox)
+            image = model.getImage().getArray()
+            image += self.randGen.random((size, size))*noiseLevel
+            for x, y, f in zip(xLoc, yLoc, flux):
+                xVals = np.exp(-np.power(np.arange(size) - x, 2.)/(2*np.power(psfSize, 2.)))
+                yVals = np.exp(-np.power(np.arange(size) - y, 2.)/(2*np.power(psfSize, 2.)))
+                image += f*np.outer(yVals, xVals)
+            imageSum += image
+            model.getVariance().getArray()[:, :] = image
+            model.mask.addMaskPlane("CLIPPED")
+            self.dcrModels.append(model)
+        maskVals = np.zeros_like(imageSum)
+        maskVals[imageSum > 5*noiseLevel] = afwImage.Mask.getPlaneBitMask('DETECTED')
+        for model in self.dcrModels:
+            model.getMask().getArray()[:, :] = maskVals
+        self.mask = self.dcrModels[0].getMask()
+
+    def makeDummyWcs(self, rotAngle, visitInfo=None):
+        """! Make a World Coordinate System object for testing.
+
+        @param[in] rotAngle: rotation of the CD matrix, East from North as a lsst.afw.geom.Angle
+        @param[in] visitInfo: lsst.afw.image.VisitInfo for the exposure.
+
+        @return wcs, a lsst.afw.geom.skyWcs.skyWcs.SkyWcs object.
+        """
+        if visitInfo is None:
```

Comment:
Why allow for this possibility? Shouldn't you provide an RA,Dec for every input?